File: server/src/main/scala/kpn/api/common/diff/IdDiffs.scala
package kpn.api.common.diff

object IdDiffs {
  def empty: IdDiffs = IdDiffs()
}

case class IdDiffs(
  removed: Seq[Long] = Seq.empty,
  added: Seq[Long] = Seq.empty,
  updated: Seq[Long] = Seq.empty
) {

  def ids: Seq[Long] = removed ++ added ++ updated

}

File: backend/api/src/main/java/kpn/api/common/diff/IdDiffs.java
package kpn.api.common.diff;

import com.google.common.collect.ImmutableList;

public record IdDiffs(
  ImmutableList<Long> removed,
  ImmutableList<Long> added,
  ImmutableList<Long> updated
) {

  public static IdDiffs empty() {
    return new IdDiffs(
      ImmutableList.of(),
      ImmutableList.of(),
      ImmutableList.of()
    );
  }

  public ImmutableList<Long> ids() {
    return ImmutableList.<Long>builder()
      .addAll(removed)
      .addAll(added)
      .addAll(updated)
      .build();
  }

  public boolean nonEmpty() {
    return !removed.isEmpty() || !added.isEmpty() || !updated.isEmpty();
  }
}
